cmd/id-exporter/main.go

```diff
@@ -66,19 +68,16 @@ func (i *idExporterResults) writeToFile(outfile string) error {
 	return ioutil.WriteFile(outfile, data, 0644)
 }
 
-// Find all registration IDs with unexpired certificates.
+// findIDs gathers all registration IDs with unexpired certificates.
 func (c idExporter) findIDs() (idExporterResults, error) {
 	var holder idExporterResults
 	_, err := c.dbMap.Select(
 		&holder,
-		`SELECT id
-		FROM registrations
-		WHERE contact != 'null' AND
-			id IN (
-				SELECT registrationID
-				FROM certificates
-				WHERE expires >= :expireCutoff
-			);`,
+		`SELECT DISTINCT r.id
+		FROM registrations AS r
+			INNER JOIN certificates AS c on c.registrationID = r.id
+		WHERE r.contact != '[]'
```

Comment:
A little searching shows that we have both `[]` and `null` in the databases for contacts. Bummer that it is so, but it is. Let's just make this instead:
```suggestion
		WHERE r.contact NOT IN ('[]', 'null')
```